addons/web/static/src/views/list/list_renderer.js

```diff
@@ -186,6 +191,45 @@ export class ListRenderer extends Component {
             this.columns = this.getActiveColumns();
             this.withHandleColumn = this.columns.some((col) => col.widget === "handle");
         });
+        onWillStart(async () => {
+            const values = this.props.list.groups
+                ? this.props.list.groups.map((g) => g.aggregates)
+                : this.props.list.records.map((r) => r.data);
+            const currencies = new Set();
+            this.fieldCurrencies = {};
+            for (const column of this.props.archInfo.columns) {
+                const field = this.props.list.fields[column.name];
+                if (
+                    (field?.type === "monetary" && field.currency_field) ||
```

Comment:
```suggestion
                if (column.type !== "field") {
                    continue;
                }
                const field = this.props.list.fields[column.name];
                if (
                    (field.type === "monetary" && field.currency_field) ||
```

`?.` makes it look like you don't know what you're doing